libcxx/include/deque

```diff
@@ -282,7 +293,18 @@ template <class _ValueType,
 #  endif
           >
 class __deque_iterator {
-  typedef _MapPointer __map_iterator;
+  using __map_iterator _LIBCPP_NODEBUG = __get_deque_map_iterator<_ValueType, _MapPointer>;
```

Comment:
Isn't this just `__rebind_pointer_t<_MapPointer, _ValueType>`?

Reply:
It isn't. As noted in the comments, what we want is `FancyPtr<const FancyPtr<_ValueType>>` here (if the iterator is const), but not `FancyPtr<_ValueType>`.